framework/platform/bpl/uci/db/bpl_db_uci.h

```diff
@@ -0,0 +1,135 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * SPDX-FileCopyrightText: 2020 the prplMesh contributors (see AUTHORS.md)
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#ifndef _BPL_DB_UCI_H_
+#define _BPL_DB_UCI_H_
+
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace beerocks {
+namespace bpl {
+namespace db {
+
+/****************************************************************************/
+/******************************* Definitions ********************************/
+/****************************************************************************/
+
+/****************************************************************************/
+/******************************* Structures *********************************/
+/****************************************************************************/
+
+/****************************************************************************/
+/******************************** Functions *********************************/
+/****************************************************************************/
+
+/**
+ * @brief Find if section exists.
+ * 
+ * @param[in] package_name name of the requested configuration file.
+ * @param[in] section_type type of the requested section.
+ * @param[in] section_name name of the requested section.
+ * @return true if section exists in db, false otherwise
+ */
+bool uci_section_exists(const std::string &package_name, const std::string &section_type,
+                        const std::string &section_name);
+
+/**
+ * @brief Add new named section.
+ * 
+ * @param[in] package_name name of the requested configuration file.
+ * @param[in] section_type type of the requested section.
+ * @param[in] section_name name of the requested section.
+ * @return true if section exists in db, false otherwise
+ */
+bool uci_add_section(const std::string &package_name, const std::string &section_type,
+                     const std::string &section_name);
+
+/**
+ * @brief Set values in section, updating as needed.
+ * An option set to an empty string will be removed from the section.
+ * An option that does not exist will be added to the section.
+ * An option that already exists will be overridden with the new value
+ * Options that exist and are not mentioned in @a params will remain unchanged.
+ * 
+ * @param[in] package_name name of the requested configuration file.
+ * @param[in] section_type type of the requested section.
+ * @param[in] section_name name of the requested section.
+ * @param[in] params unordered_map containing the params the requested section, to be set.
```

Comment:
```suggestion
 * @param[in] options unordered_map containing a key/value pair of parameters to set.
```

you might want to review rest of Doxygen here if you choose to accept this suggestion

Reply:
Updated Doxygens throughout the file